src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -27,4 +28,33 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null) {
+            return false;
+        }
+        if (obj.getClass().equals(Rectangle.class)) {
+            Rectangle rectangle = (Rectangle) obj;
+            return (color != null && color.equals(rectangle.color)
+                    || (rectangle.color == color))
+                    && (width != null && width.equals(rectangle.width)
+                    || (rectangle.width == width))
+                    && (length != null && length.equals(rectangle.length)
+                    || (rectangle.length == length));
+        }
+        return false;
+    }
+
+    @Override
+    public int hashCode() {
+        int result = 0;
+        result *= HASHCODENUMBER + (width != null ? width.hashCode() : 0);
+        result *= HASHCODENUMBER + (color != null ? color.hashCode() : 0);
+        result *= HASHCODENUMBER + (length != null ? length.hashCode() : 0);
```

Comment:
we are multiplying `result` on each line
`result` value is 0
what do we get if we multiply any value by 0?

please reply to this comment below

Reply:
We'll get eternal zero. I did not initialize correctly result.  I had to initialize "result" with any positive prime number.